### PR TITLE
feat: Initialize lists with map

### PR DESCRIPTION
**Pull request by webmaster128**
This uses `map` to initialize lists, such that all cases can be handled within one assignment. `if (object.ids !== undefined && object.ids !== null) {` was replaced with `object.ids ?? []` to save nesting depth and code size.

### CODE REVIEW

**Comment by webmaster128** [comment]
Any idea why CI is failing here? The problematic "Example 6" was added here in #370. The other integration/*/google/protobuf/timestamp.ts alle have the older version of the comment. Why did this not fail in the PR #370 or on main but now?

**Review by stephenh — APPROVED** [review]

**Comment by stephenh** [comment]
@webmaster128 yeah, I'm not really sure how that `timestamp.ts` change got through that other PR... like it was only a month or so ago, and I haven't changed any of the lint/CI setup in quite awhile.

I merged #388 so am personally pretty willing to procrastinate on figuring out true root cause + better yet moving to a better `protoc` setup. :-) But if that's something you'd like to improve, that'd be great, we definitely need something more deterministic than the current setup.

This PR also looks great! Thanks!

File: src/main.ts [hunk 1]
```diff
@@ -1249,7 +1225,7 @@ function generateFromPartial(ctx: Context, fullName: string, messageDesc: Descri
       chunks.push(code`message.${fieldName} = ${readSnippet(`object.${fieldName}`)};`);
       chunks.push(code`} else {`);
       const fallback = isWithinOneOf(field) ? 'undefined' : defaultValue(ctx, field);
-      chunks.push(code`message.${fieldName} = ${fallback}`);
+      chunks.push(code`message.${fieldName} = ${fallback};`);
```
Comment (webmaster128) [inline src/main.ts:1228]: This change is unrelated and simply here to increase consistency between `generateFromPartial` and `generateFromJson`.

**Comment by webmaster128** [comment]
Rebased. When it passes CI, I'm happy.
